### PR TITLE
Update aws sdk

### PR DESCRIPTION
**Pull request by famarting**
https://github.com/aws/aws-msk-iam-sasl-signer-go/issues/16

### CODE REVIEW

**Comment by yaron2** [comment]
cc @agarwal1510

File: signer/msk_auth_token_provider.go [hunk 1]
```diff
@@ -16,7 +16,7 @@ import (
 	"time"
 
 	"github.com/aws/aws-sdk-go-v2/aws"
-	"github.com/aws/aws-sdk-go-v2/aws/signer/v4"
+	v4 "github.com/aws/aws-sdk-go-v2/aws/signer/v4"
```
Comment (agarwal1510) [inline signer/msk_auth_token_provider.go:19]: Do we need this v4 here?

Comment (famarting) [reply signer/msk_auth_token_provider.go:19]: linter added it

**Review by agarwal1510 — APPROVED** [review]

**Comment by agarwal1510** [comment]
@famarting Go tests are failing with error: `unknown directive: toolchain`. Can you verify the fix and update your PR?

**Comment by famarting** [comment]
can you trigger CI again @agarwal1510

**Comment by yaron2** [comment]
Ping @agarwal1510

**Review by agarwal1510 — APPROVED** [review]

**Comment by famarting** [comment]
@agarwal1510 can you trigger CI again?

I had to update the supported go versions following https://github.com/aws/aws-sdk-go-v2/blob/main/.github/workflows/go.yml

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -16,7 +16,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest, macos-latest, windows-latest]
-        go-version: [1.19, "1.20"]
+        go-version: ["1.21", "1.22", "1.23"]
```
Comment (agarwal1510) [inline .github/workflows/go.yml:19]: We want to support Go 1.20 which is also supported by AWS SDK. Please change the min to 1.20 in the PR.

**Review by agarwal1510 — APPROVED** [review]

File: .github/workflows/go.yml (hunk 2 above)
Comment (agarwal1510) [reply .github/workflows/go.yml:19]: Was looking at outdated documentation, the library supports 1.21 as the minimum version. Approved.

**Comment by famarting** [comment]
Thank you

**Pull request merged**